Contact Summary address: make accordion more accessible

highfalutin opened the pull request:
Overview
----------------------------------------
Toward [user-interface#60](https://lab.civicrm.org/dev/user-interface/-/issues/60) / [core#3294](https://lab.civicrm.org/dev/core/-/issues/3294) -- replaces the [last remaining instance of .crm-collapsible](https://lab.civicrm.org/dev/user-interface/-/issues/67#note_160008), a less-accessible accordion pattern, with `<details><summary>`

Before
----------------------------------------
`.crm-collapsible` plus JS controls

After
----------------------------------------
`<details><summary>`, with css class to play well with existing JS

highfalutin committed:
change accordion to details+summary

highfalutin commented:
@herbdool this is currently broken due to some interactions with JS -- do you want to investigate?

highfalutin committed:
allow details to be opened without triggering inline edit

highfalutin commented:
Actually @herbdool I figured out the js issue. The class `.collapsible-title` needed to be present on the `<summary>` element so that clicks wouldn't trigger online edit (see [the relevant js](https://github.com/lemniscus/civicrm-core/blob/6080cddd7fee7fcf68d841295d6962c461ee2033/templates/CRM/Contact/Page/View/Summary.js#L206)).

@vingle can you help us to make it look good? See [demo](http://core-29594-8hdk5.test-1.civicrm.org:8001/civicrm/contact/view?reset=1&cid=4)

herbdool commented:
@highfalutin the PR works for me. (Just tested while waiting for the train to start :wink:)

I'm wondering if we want to add `<summary>` to that JS so `.collapsible-title` doesn't need to be added? Would this work:

```
 .on('mousedown', '.crm-inline-edit:not(.form) a, .crm-inline-edit:not(.form) .crm-accordion-header, .crm-inline-edit:not(.form) summary', function(event) {
```

I'm guessing there aren't any situations when clicking on an accordion *should* force in-place editing.

highfalutin commented:
@herbdool that's a great idea. I think maybe it belongs in a separate PR, where we can wipe out [all instances of the `collapsible-title` class and the references to it](https://github.com/search?q=repo%3Acivicrm%2Fcivicrm-core+collapsible-title&type=code).

vingle committed:
Give accordion light utility class

Other custom field accordions on the contact dashboard use `.crm-accordion-light`.

vingle commented:
That works really well, thanks for persevering to figure it out. Have added a small styling PR [here](https://github.com/lemniscus/civicrm-core/pull/1).

highfalutin committed:
Merge pull request #1 from vingle/patch-23

Gives inline accordion the light utility class
